src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -7,50 +7,87 @@
 import io.prometheus.client.exporter.HTTPServer;
 import org.apache.kafka.common.config.AbstractConfig;
 import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.net.BindException;
+import java.net.InetSocketAddress;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
- * Configuration for the reporter implementations.
+ /**
+ * Configuration for the PrometheusMetricsReporter implementation.
  */
 public class PrometheusMetricsReporterConfig extends AbstractConfig {
 
     private static final Logger LOG = LoggerFactory.getLogger(PrometheusMetricsReporterConfig.class.getName());
 
+    /**
+     * Configuration prefix for Prometheus Metrics Reporter.
+     */
     public static final String CONFIG_PREFIX = "prometheus.metrics.reporter.";
 
-    public static final String PORT_CONFIG = CONFIG_PREFIX + "port";
-    public static final int PORT_CONFIG_DEFAULT = 8080;
-    public static final String PORT_CONFIG_DOC = "The HTTP port to expose the metrics.";
+    /**
+     * Configuration key for the listener to expose the metrics.
+     */
+    public static final String LISTENER_CONFIG = CONFIG_PREFIX + "listener";
 
+    /**
+     * Default value for the listener configuration.
+     */
+    public static final String LISTENER_CONFIG_DEFAULT = "http://:8080";
+
+    /**
+     * Documentation for the listener configuration.
+     */
+    public static final String LISTENER_CONFIG_DOC = "The HTTP listener to expose the metrics.";
```

Comment:
All the `DOC` fields could be private too

Reply:
Changed to private

Reply:
This field is still `public`